cherab/tools/observers/spectroscopic.py

```diff
@@ -440,28 +444,29 @@ def sight_lines(self):
     @sight_lines.setter
     def sight_lines(self, value):
 
-        if not isinstance(value, list):
-            raise TypeError("The sightlines attribute of LineOfSightGroup must be a list of SpectroscopicSightLines.")
+        if not isinstance(value, (list, tuple)):
+            raise TypeError("The sight_lines attribute of LineOfSightGroup must be a list or tuple of SpectroscopicSightLines.")
 
         for sight_line in value:
             if not isinstance(sight_line, SpectroscopicSightLine):
-                raise TypeError("The sightlines attribute of LineOfSightGroup must be a list of "
+                raise TypeError("The sight_lines attribute of LineOfSightGroup must be a list or tuple of "
                                 "SpectroscopicSightLines. Value {} is not a SpectroscopicSightLine.".format(sight_line))
 
         # Prevent external changes being made to this list
-        value = value.copy()
         for sight_line in value:
             sight_line.parent = self
 
-        self._sight_lines = value
+        self._sight_lines = tuple(value)
 
     def add_sight_line(self, sight_line):
 
         if not isinstance(sight_line, SpectroscopicSightLine):
-            raise TypeError("The sightline argument must be of type SpectroscopicSightLine.")
+            raise TypeError("The sight_line argument must be of type SpectroscopicSightLine.")
 
         sight_line.parent = self
-        self._sight_lines.append(sight_line)
+        new_sight_lines = list(self._sight_lines)
+        new_sight_lines.append(sight_line)
+        self._sight_lines = tuple(new_sight_lines)
```

Comment:
There is a cleaner way to do this:
```python
self._sight_lines = self._sight_lines + (sight_line,)
```
Tuples have an `__add__` method which will return a new tuple with the operand appended on the end.

Reply:
Thanks. I fixed this.